src/Uno.UI.Runtime.Skia.Gtk/GLRenderSurface.cs

```diff
@@ -125,9 +127,16 @@ private void UnoGLDrawingArea_Render(object o, RenderArgs args)
 			using (new SKAutoCanvasRestore(canvas, true))
 			{
 				canvas.Clear(SKColors.White);
+
+				if (_dpi != null)
+				{
+					canvas.SetMatrix(SKMatrix.CreateScale((float)dpi, (float)dpi));
+				}
 				canvas.Translate(new SKPoint(0, GuardBand));
 
 				WUX.Window.Current.Compositor.Render(_surface);
+
+				canvas.SetMatrix(SKMatrix.CreateScale((float)1 / dpi, (float)1 / dpi));
```

Comment:
I do not think this is required. The call to `SKAutoCanvasRestore.Dispose()` should reset the matrix of the canvas.

Reply:
Fixed in recent changes